src/Forms/GridField/FormAction/StateStore.php

```diff
@@ -0,0 +1,30 @@
+<?php
+namespace SilverStripe\Forms\GridField\FormAction;
+
+use SilverStripe\Control\HTTPRequest;
+
+interface StateStore
+{
+    /**
+     * @param HTTPRequest $request
+     */
+    public function __construct(HTTPRequest $request);
```

Comment:
I'm in two minds about whether it's a good idea to have the constructor in this interface

Reply:
I've just read this and now believe we should remove the constructor: https://www.alainschlesser.com/including-constructor-interface/

Reply:
Yeah at the time I wasn't really happy with this code either. I need the request in both implementations of the interface and I don't think it's a good pattern to fetch the request from global state (`Controller::curr` or `Injector`). I'll just suck it up though; remove the constructor and use `Controller::curr`.

Reply:
You could always give it a setter and use injector calls to provide it

Reply:
Or put the constructor in an abstract parent class - separating the implementation from the interface entirely